crates/transaction-pool/src/blobstore/disk.rs

```diff
@@ -62,12 +62,16 @@ impl BlobStore for DiskFileBlobStore {
     }
 
     fn delete(&self, tx: B256) -> Result<(), BlobStoreError> {
-        self.inner.txs_to_delete.write().insert(tx);
+        if self.inner.contains(tx)? {
+            self.inner.txs_to_delete.write().insert(tx);
+        }
         Ok(())
     }
 
     fn delete_all(&self, txs: Vec<B256>) -> Result<(), BlobStoreError> {
-        self.inner.txs_to_delete.write().extend(txs);
+        for tx in txs {
+            let _ = self.delete(tx);
+        }
```

Comment:
we can optimize this slightly by doing a specialized impl here,
we only want to acquire the locks once

so we should add another function to inner that is

```rust
fn retain_existing(&self, v: &mut Vec) {
  // acquire lock once then do v.retain
  // then retain if exists on disk
}
```

https://github.com/paradigmxyz/reth/blob/cf7698ab035bcb8209ea6cd422feace7ce37fc47/crates/transaction-pool/src/blobstore/disk.rs#L226-L232